language/stdlib/modules/libra_configs.move

```diff
@@ -1,32 +1,101 @@
 address 0x0:
 module LibraConfig {
     use 0x0::Transaction;
-    use 0x0::LibraSystem;
+    use 0x0::LibraAccount;
+    use 0x0::LibraTimestamp;
 
     // A generic singleton resource that holds a value of a specific type.
     resource struct T<Config: copyable> { payload: Config }
 
-    // Set a config item to a new value and trigger a reconfiguration.
-    // This can only be invoked by the Association address
-    public fun set<Config: copyable>(payload: Config) acquires T {
+    struct NewEpochEvent {
+        epoch: u64,
+    }
+
+    resource struct Configuration {
+        epoch: u64,
+        last_reconfiguration_time: u64,
+        events: LibraAccount::EventHandle<NewEpochEvent>,
+    }
+
+    // This can only be invoked by the Association address, and only a single time.
+    // Currently, it is invoked in the genesis transaction
+    public fun initialize_configuration() {
+        let sender = Transaction::sender();
+        // Only callable by the Association address for now.
+        Transaction::assert(sender == 0xA550C18, 1);
+
+        move_to_sender<Configuration>(Configuration {
+            epoch: 0,
+            last_reconfiguration_time: 0,
+            events: LibraAccount::new_event_handle<NewEpochEvent>(),
+        });
+    }
+
+    // We can't return reference here, this returns a copy of the config value.
+    public fun get<Config: copyable>(): Config acquires T {
         let sender = Transaction::sender();
 
         // Only callable by the Association address for now.
         Transaction::assert(sender == 0xA550C18, 1);
 
+        Transaction::assert(::exists<T<Config>>(sender), 24);
+        let t_ref = borrow_global_mut<T<Config>>(sender);
+
+        *&t_ref.payload
+    }
+
+    // Set a config item to a new value and trigger a reconfiguration.
+    // This can only be invoked by the Association address
+    public fun set<Config: copyable>(payload: Config) acquires T, Configuration {
+        let sender = Transaction::sender();
+
```

Comment:
here you need the permission checks

Reply:
permission check is moved to reconfigure function

Reply:
Oh, I see, you are only allowed to call set if you have this resource... asserting that the resource exists (49) is interesting... what happens if we delete it, have no such resource, and call config.payload

Reply:
it'll abort, we don't support delete right now, but I imagine if we delete it, it has to go through `publish_new_config`